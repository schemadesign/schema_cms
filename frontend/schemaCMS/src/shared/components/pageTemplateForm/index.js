export { PageTemplateForm } from './pageTemplateForm.component';
