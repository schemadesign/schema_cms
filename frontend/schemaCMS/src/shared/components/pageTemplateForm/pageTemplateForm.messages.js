/* eslint-disable max-len */
import { defineMessages } from 'react-intl';
import {
  PAGE_TEMPLATES_NAME,
  PAGE_TEMPLATES_IS_AVAILABLE,
  PAGE_TEMPLATES_ALLOW_EDIT,
} from '../../../modules/pageTemplates/pageTemplates.constants';

export default defineMessages({
  blocks: {
    id: 'shared.components.pageTemplateForm.blocks',
    defaultMessage: 'Block',
  },
  [`${PAGE_TEMPLATES_NAME}Placeholder`]: {
    id: `shared.components.pageTemplateForm.${PAGE_TEMPLATES_NAME}Placeholder`,
    defaultMessage: 'Name',
  },
  [PAGE_TEMPLATES_NAME]: {
    id: `shared.components.pageTemplateForm.${PAGE_TEMPLATES_NAME}`,
    defaultMessage: 'Name',
  },
  [PAGE_TEMPLATES_IS_AVAILABLE]: {
    id: `shared.components.pageTemplateForm.${PAGE_TEMPLATES_IS_AVAILABLE}`,
    defaultMessage: 'Make it Available',
  },
  [PAGE_TEMPLATES_ALLOW_EDIT]: {
    id: `shared.components.pageTemplateForm.${PAGE_TEMPLATES_ALLOW_EDIT}`,
    defaultMessage: 'Allow editing the Page Template',
  },
  availableForEditors: {
    id: 'shared.components.pageTemplateForm.availableForEditors',
    defaultMessage: 'This Template is currently {negative}available for Editors',
  },
  copyConfirmTitle: {
    id: 'shared.components.pageTemplateForm.copyConfirmTitle',
    defaultMessage: 'There are unsaved data. Do you want to copy without it?',
  },
  cancelCopy: {
    id: 'shared.components.pageTemplateForm.cancelCopy',
    defaultMessage: 'Cancel',
  },
  confirmCopy: {
    id: 'shared.components.pageTemplateForm.confirmCopy',
    defaultMessage: 'Copy',
  },
});
