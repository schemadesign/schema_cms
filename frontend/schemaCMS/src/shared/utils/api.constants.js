const API_PREFIX = process.env.REACT_APP_BASE_API_URL;

export const AUTH_PATH = `${API_PREFIX}/auth/login/auth0`;
export const TOKEN_PATH = '/auth/token/';
export const USER_PATH = '/user/';
