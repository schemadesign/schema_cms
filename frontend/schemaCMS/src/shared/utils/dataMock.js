export const tableData = [
  {
    'First Name': 'George',
    'Last Name': 'Washington',
    'Full Name': 'George Washington',
    Birthplace: 'Westmoreland County',
    'State of Birth': 'VA',
    Birthday: '02/22/1732',
    Url: 'some.url.com',
  },
  {
    'First Name': 'John',
    'Last Name': 'Adams',
    'Full Name': 'John Adams',
    Birthplace: 'Braintree',
    'State of Birth': 'MA',
    Birthday: '10/30/1735',
    Url: 'some.url.com',
  },
  {
    'First Name': 'Thomas',
    'Last Name': 'Jefferson',
    'Full Name': 'Thomas Jefferson',
    Birthplace: 'Shadwell',
    'State of Birth': 'VA',
    Birthday: '04/13/1743',
    Url: 'some.url.com',
  },
  {
    'First Name': 'James',
    'Last Name': 'Madison',
    'Full Name': 'James Madison',
    Birthplace: 'Port Conway',
    'State of Birth': 'VA',
    Birthday: '03/16/1751',
    Url: 'some.url.com',
  },
  {
    'First Name': 'James',
    'Last Name': 'Monroe',
    'Full Name': 'James Monroe',
    Birthplace: 'Monroe Hall',
    'State of Birth': 'VA',
    Birthday: '04/28/1758',
    Url: 'some.url.com',
  },
];

export const tableFields = {
  'First Name': {},
  'Last Name': {},
  'Full Name': {},
  Birthplace: {},
  'State of Birth': {},
  Birthday: {},
  Url: {},
};
