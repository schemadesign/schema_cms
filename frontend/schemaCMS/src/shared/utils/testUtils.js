import { Router } from 'react-router';
import { IntlProvider } from 'react-intl';
import { act, create } from 'react-test-renderer';
import React from 'react';
import { ThemeProvider } from 'styled-components';
import { Theme } from 'schemaUI';
import PropTypes from 'prop-types';

import { DEFAULT_LOCALE, translationMessages } from '../../i18n';
import browserHistory from './history';

const ProvidersWrapper = ({ children }) => (
  <Router history={browserHistory}>
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <ThemeProvider theme={Theme.dark}>{children}</ThemeProvider>
    </IntlProvider>
  </Router>
);

ProvidersWrapper.propTypes = {
  children: PropTypes.node.isRequired,
};

export const makeContextRenderer = async component => {
  let wrapper = null;

  await act(async () => {
    wrapper = await create(<ProvidersWrapper>{component}</ProvidersWrapper>);
  });

  return wrapper;
};
