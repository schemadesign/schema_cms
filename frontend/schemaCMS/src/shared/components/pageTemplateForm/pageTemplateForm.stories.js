import React from 'react';
import { storiesOf } from '@storybook/react';

import { PageTemplateForm } from './pageTemplateForm.component';
import { blockTemplates } from '../../../modules/blockTemplates/blockTemplates.mocks';
import { withTheme } from '../../../.storybook/decorators';
import { pageTemplate } from '../../../modules/pageTemplates/pageTemplates.mocks';

export const defaultProps = {
  handleChange: Function.prototype,
  setFieldValue: Function.prototype,
  setValues: Function.prototype,
  setRemoveModalOpen: Function.prototype,
  blockTemplates,
  values: pageTemplate,
  isValid: true,
  title: <span>title</span>,
};

storiesOf('PageTemplateForm', module)
  .addDecorator(withTheme())
  .add('Default', () => <PageTemplateForm {...defaultProps} />);
