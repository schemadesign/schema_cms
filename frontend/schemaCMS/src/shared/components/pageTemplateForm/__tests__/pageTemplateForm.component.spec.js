import React from 'react';

import { PageTemplateForm } from '../pageTemplateForm.component';
import { defaultProps } from '../pageTemplateForm.stories';
import { makeContextRenderer } from '../../../utils/testUtils';

describe('PageTemplateForm: Component', () => {
  const render = props => makeContextRenderer(<PageTemplateForm {...defaultProps} {...props} />);

  it('should render correctly', async () => {
    const wrapper = await render();
    expect(wrapper).toMatchSnapshot();
  });

  it('should add new block', async () => {
    jest.spyOn(defaultProps, 'setFieldValue');
    const wrapper = await render();
    wrapper.root.findAllByProps({ id: 'createBlock' })[0].props.onClick();

    expect(defaultProps.setFieldValue).toHaveBeenCalledWith('blocks', [
      { autoOpen: true, key: 1575162000000, name: '', block: '' },
      { id: 1, key: 1, name: 'block name', block: 1 },
    ]);
  });

  it('should call setRemoveModalOpen', async () => {
    jest.spyOn(defaultProps, 'setRemoveModalOpen');
    const wrapper = await render();
    wrapper.root.findByProps({ id: 'removePage' }).props.onClick();

    expect(defaultProps.setRemoveModalOpen).toHaveBeenCalledWith(true);
  });
});
