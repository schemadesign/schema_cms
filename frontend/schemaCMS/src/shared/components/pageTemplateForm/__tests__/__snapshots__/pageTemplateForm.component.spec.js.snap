// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTemplateForm: Component should render correctly 1`] = `
.c0 {
  display: none;
}

.c1 {
  margin-right: 20px;
  overflow: hidden;
}

.c2 {
  color: #FFF;
}

.c3 {
  color: #FFF;
}

.c23 {
  margin-top: 30px;
  border-top: 2px solid #2F2F36;
  padding-top: 15px;
}

.c24 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-bottom: 15px;
  position: relative;
}

.c27 {
  padding-right: 10px;
}

.c26 {
  padding-left: 10px;
  max-width: 100%;
}

.c25 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 100%;
  padding-right: 10px;
}

.c29 {
  cursor: pointer;
  position: absolute;
  right: 0;
  top: 0;
}

.c8 {
  cursor: pointer;
}

.c28 {
  color: #71737e;
  min-width: 100%;
}

.c4 {
  position: relative;
}

.c7 {
  position: absolute;
  right: 0;
  top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c11 {
  position: relative;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: end;
  -ms-flex-pack: end;
  justify-content: end;
}

.c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: move;
  touch-action: none;
}

.c5 {
  position: relative;
  max-width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
}

.c12 {
  position: relative;
}

.c13 {
  margin-top: 0;
}

.c6 {
  position: relative;
  top: 0;
  left: 0;
  color: #f21e29;
  font-size: 14px;
  font-weight: normal;
}

.c14 {
  color: #71737e;
  text-align: center;
  padding-bottom: 20px;
  min-height: 43px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c15 {
  width: 33%;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  border-top: 1px solid #FFF;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c20 {
  padding-left: 0;
  width: calc(100% - 81px);
}

.c21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-bottom: 30px;
  padding-right: 30px;
  position: relative;
  font-size: 18px;
  cursor: pointer;
  padding-bottom: 0;
}

.c21 label {
  margin-bottom: 2px;
}

.c22 {
  position: absolute;
  right: 0;
  top: 5px;
  top: 50%;
  right: 5px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  font-size: 0;
}

.c9 {
  cursor: pointer;
  position: relative;
}

@media (min-width:1050px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    margin: 60px 0 40px 0;
    min-height: 63px;
  }
}

@media (min-width:1050px) {
  .c10 {
    display: block;
  }
}

@media (min-width:480px) {
  .c24 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (min-width:480px) {
  .c25 {
    padding-right: 40px;
  }
}

@media (min-width:1050px) {
  .c11 {
    display: none;
  }
}

@media (min-width:1050px) {
  .c16 {
    display: none;
  }
}

@media (min-width:1050px) {
  .c13 {
    margin-top: 0;
  }
}

@media (min-width:1050px) {
  .c14 {
    margin-top: -35px;
    padding-bottom: 40px;
  }
}

@media (min-width:768px) {

}

<div
  className=""
>
  <div
    className="c0"
  >
    <div
      className="c1"
    >
      <h2
        className="c2"
        id="desktopHeaderTitle"
        style={
          Object {
            "fontFamily": "Inter, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": "24px",
            "fontWeight": 100,
            "margin": 0,
          }
        }
      >
        <span>
          title
        </span>
      </h2>
      <h1
        className="c3"
        id="desktopHeaderSubtitle"
        style={
          Object {
            "fontFamily": "Inter, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": "24px",
            "fontWeight": 600,
            "letterSpacing": "-0.12px",
            "lineHeight": 1.25,
            "margin": 0,
          }
        }
      >
        <div
          className="c4"
        >
          <div
            className="c5"
          >
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                  "maxWidth": "100%",
                  "paddingBottom": 0,
                  "paddingRight": 40,
                  "position": "static",
                }
              }
            >
              <input
                id="name"
                name="name"
                onChange={[Function]}
                placeholder="Name"
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "border": "none",
                    "color": "#FFF",
                    "fontFamily": "inherit",
                    "fontSize": 24,
                    "fontWeight": 600,
                    "letterSpacing": "normal",
                    "lineHeight": "24px",
                    "maxWidth": "100%",
                    "minWidth": 60,
                    "outline": "none",
                    "padding": 0,
                    "width": 80,
                  }
                }
                type="text"
                value="Page template"
              />
              <div
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "border": "none",
                    "color": "#FFF",
                    "fontFamily": "inherit",
                    "fontSize": 24,
                    "fontWeight": 600,
                    "letterSpacing": "normal",
                    "lineHeight": "24px",
                    "maxWidth": "100%",
                    "minWidth": 60,
                    "opacity": 0,
                    "outline": "none",
                    "padding": 0,
                    "position": "fixed",
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                Page template
              </div>
              <div
                style={
                  Object {
                    "position": "absolute",
                    "right": 0,
                    "top": 2,
                  }
                }
              >
                <label
                  htmlFor="name"
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                />
              </div>
            </div>
            <div
              className="c6"
              id="nameError"
            />
          </div>
          <div
            className="c7"
          >
            <label
              className="c8"
              htmlFor="name"
            >
              <svg
                style={
                  Object {
                    "fill": "#FFF",
                    "height": "30px",
                    "width": "30px",
                  }
                }
                viewBox="0 0 30 30"
              >
                <path
                  d="M21.789 10.92a.724.724 0 000-1.022l-1.687-1.687a.724.724 0 00-1.022 0l-1.322 1.322 2.709 2.709 1.322-1.322zM9 18.291V21h2.709l7.992-7.992-2.709-2.71L9 18.292z"
                  fillRule="evenodd"
                />
              </svg>
            </label>
            <div
              className="c9"
              id="pageTemplateCopyButton"
              onClick={[Function]}
            >
              <svg
                style={
                  Object {
                    "fill": "#FFF",
                    "height": "20px",
                    "width": "20px",
                  }
                }
                viewBox="0 0 512 512"
              >
                <path
                  d="M464 64h-48V48c0-26.51-21.49-48-48-48H48C21.49 0 0 21.49 0 48v320c0 26.51 21.49 48 48 48h16v48c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V112c0-26.51-21.49-48-48-48zM64 112v272H48c-8.837 0-16-7.163-16-16V48c0-8.837 7.163-16 16-16h320c8.837 0 16 7.163 16 16v16H112c-26.51 0-48 21.49-48 48zm416 352c0 8.837-7.163 16-16 16H112c-8.837 0-16-7.163-16-16V112c0-8.837 7.163-16 16-16h352c8.837 0 16 7.163 16 16v352z"
                />
              </svg>
            </div>
          </div>
        </div>
      </h1>
    </div>
    <div
      className=""
    >
      <div
        className="c10"
      >
        <button
          disabled={false}
          id="createBlock"
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFF",
              "border": "none",
              "borderRadius": "30px",
              "color": "#000",
              "cursor": "pointer",
              "display": "flex",
              "fill": "#FFF",
              "fontSize": "20px",
              "height": "60px",
              "justifyContent": "center",
              "margin": "0",
              "minHeight": "48px",
              "outline": "none",
              "padding": "0",
              "transition": "background 0.25s ease-in-out, color 0.25s ease-in-out",
              "width": "60px",
            }
          }
          type="button"
        >
          <svg
            style={
              Object {
                "fill": "#000",
                "height": "60px",
                "shapeRendering": "crispEdges",
                "width": "60px",
              }
            }
            viewBox="0 0 60 60"
          >
            <path
              d="M32 15v13h13v4H32v13h-4V32H15v-4h13V15h4z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
  <div
    className="c11"
  >
    <div
      className="c12"
    >
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "maxWidth": "100%",
            "paddingBottom": "30px",
            "position": "relative",
          }
        }
      >
        <label
          htmlFor="name"
          style={
            Object {
              "border": "none",
              "borderTop": "2px solid #2F2F36",
              "color": "#71737e",
              "display": "block",
              "fontSize": "14px",
              "outline": "none",
              "paddingTop": "8px",
              "width": "100%",
            }
          }
        >
          Name
        </label>
        <input
          id="name"
          name="name"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#FFF",
              "fontFamily": "inherit",
              "fontSize": "18px",
              "fontWeight": "inherit",
              "letterSpacing": "normal",
              "lineHeight": "24px",
              "maxWidth": "100%",
              "minWidth": 60,
              "outline": "none",
              "padding": 0,
            }
          }
          type="text"
          value="Page template"
        />
        <div
          style={
            Object {
              "position": "absolute",
              "right": 0,
              "top": 2,
            }
          }
        >
          <label
            htmlFor="name"
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <div
              className="c13"
            >
              <svg
                style={
                  Object {
                    "fill": "#FFF",
                    "height": "30px",
                    "width": "30px",
                  }
                }
                viewBox="0 0 30 30"
              >
                <path
                  d="M21.789 10.92a.724.724 0 000-1.022l-1.687-1.687a.724.724 0 00-1.022 0l-1.322 1.322 2.709 2.709 1.322-1.322zM9 18.291V21h2.709l7.992-7.992-2.709-2.71L9 18.292z"
                  fillRule="evenodd"
                />
              </svg>
            </div>
          </label>
        </div>
      </div>
      <div
        className="c6"
        id="nameError"
      />
    </div>
  </div>
  <div
    className="c14"
  >
    <div
      className="c15"
      id="tagsCounter"
    />
    1 Block
    <div
      className="c15"
    >
      <div
        className="c16"
      >
        <div
          className="c10"
        >
          <button
            disabled={false}
            id="createBlock"
            onClick={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFF",
                "border": "none",
                "borderRadius": "30px",
                "color": "#000",
                "cursor": "pointer",
                "display": "flex",
                "fill": "#FFF",
                "fontSize": "20px",
                "height": 40,
                "justifyContent": "center",
                "margin": "0",
                "minHeight": 40,
                "outline": "none",
                "padding": "0",
                "transition": "background 0.25s ease-in-out, color 0.25s ease-in-out",
                "width": 40,
              }
            }
            type="button"
          >
            <svg
              style={
                Object {
                  "fill": "#000",
                  "height": 40,
                  "minHeight": 40,
                  "shapeRendering": "crispEdges",
                  "width": 40,
                }
              }
              viewBox="0 0 60 60"
            >
              <path
                d="M32 15v13h13v4H32v13h-4V32H15v-4h13V15h4z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
  <span>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "space-between",
          "opacity": 1,
          "position": "relative",
          "transform": "translate3d(0, 0, 0)",
          "width": "100%",
          "zIndex": 1,
        }
      }
    >
      <div
        className="c17"
      >
        <div
          className="c18"
        >
          <div>
            <div
              className="c19"
            >
              <svg
                style={
                  Object {
                    "fill": "#FFF",
                    "height": "60px",
                    "marginRight": 20,
                    "width": 10,
                  }
                }
                viewBox="0 0 10 60"
              >
                <path
                  d="M5 37a3 3 0 110 6 3 3 0 010-6zm0-10a3 3 0 110 6 3 3 0 010-6zm0-10a3 3 0 110 6 3 3 0 010-6z"
                  fillRule="evenodd"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="c20"
        >
          <div
            className="c21"
            onClick={[Function]}
          >
            <input
              id="blocks.0.block"
              name="blocks.0.block"
              type="hidden"
              value={1}
            />
            <div
              style={
                Object {
                  "color": "#FFF",
                  "display": "flex",
                  "width": "100%",
                }
              }
            >
              <div
                id="blockTypeSelect"
                style={
                  Object {
                    "cursor": "pointer",
                    "display": "flex",
                    "position": "relative",
                    "width": "100%",
                  }
                }
              >
                <div
                  onClick={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                      "display": "flex",
                      "width": "100%",
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "cursor": "pointer",
                        "overflow": "hidden",
                        "textOverflow": "ellipsis",
                        "whiteSpace": "nowrap",
                        "width": "100%",
                      }
                    }
                  >
                    block name
                  </div>
                </div>
                <div
                  style={
                    Object {
                      "WebkitOverflowScrolling": "touch",
                      "backgroundColor": "#1d1d20",
                      "display": "none",
                      "flexDirection": "column",
                      "left": "0",
                      "maxHeight": "184px",
                      "overflowY": "hidden",
                      "position": "absolute",
                      "top": "calc(100% + 5px)",
                      "width": "100%",
                      "zIndex": 99,
                    }
                  }
                >
                  <div
                    id="select-item-0"
                    onClick={[Function]}
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                    style={
                      Object {
                        "backgroundColor": "#FFF",
                        "border": "1px solid  #2F2F36",
                        "borderTop": "1px solid  #2F2F36",
                        "color": "#000",
                        "cursor": "pointer",
                        "overflow": "hidden",
                        "padding": "20px 10px",
                        "textOverflow": "ellipsis",
                        "whiteSpace": "nowrap",
                      }
                    }
                  >
                    block name
                  </div>
                  <div
                    id="select-item-1"
                    onClick={[Function]}
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                    style={
                      Object {
                        "border": "1px solid  #2F2F36",
                        "borderTop": "none",
                        "color": "#71737e",
                        "cursor": "pointer",
                        "overflow": "hidden",
                        "padding": "20px 10px",
                        "textOverflow": "ellipsis",
                        "whiteSpace": "nowrap",
                      }
                    }
                  >
                    block name 2
                  </div>
                  <div
                    id="select-item-2"
                    onClick={[Function]}
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                    style={
                      Object {
                        "border": "1px solid  #2F2F36",
                        "borderTop": "none",
                        "color": "#71737e",
                        "cursor": "pointer",
                        "overflow": "hidden",
                        "padding": "20px 10px",
                        "textOverflow": "ellipsis",
                        "whiteSpace": "nowrap",
                      }
                    }
                  >
                    block name 3
                  </div>
                </div>
              </div>
            </div>
            <div
              className="c22"
            >
              <svg
                style={
                  Object {
                    "fill": "#FFF",
                    "height": "30px",
                    "width": "30px",
                  }
                }
                viewBox="0 0 30 30"
              >
                <path
                  d="M15.117 18L9 12h12z"
                  fillRule="evenodd"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="c18"
        >
          <svg
            style={
              Object {
                "fill": "#FFF",
                "height": "30px",
                "width": "30px",
              }
            }
            viewBox="0 0 30 30"
          >
            <path
              d="M21.789 10.92a.724.724 0 000-1.022l-1.687-1.687a.724.724 0 00-1.022 0l-1.322 1.322 2.709 2.709 1.322-1.322zM9 18.291V21h2.709l7.992-7.992-2.709-2.71L9 18.292z"
              fillRule="evenodd"
            />
          </svg>
          <svg
            id="removeBlock-0"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "fill": "#FFF",
                "height": 16,
                "margin": "-3px 0 0 5px",
                "width": 16,
              }
            }
            viewBox="-47 0 512 512"
          >
            <path
              d="M416.875 114.441L405.57 80.555a31.527 31.527 0 00-29.941-21.578h-95.012V28.043C280.617 12.582 268.047 0 252.59 0h-87.008c-15.453 0-28.027 12.582-28.027 28.043v30.934H42.547a31.528 31.528 0 00-29.945 21.578L1.297 114.44a25.426 25.426 0 003.484 22.856 25.427 25.427 0 0020.578 10.539h11.817L63.184 469.44C65.117 493.305 85.367 512 109.293 512h204.863c23.922 0 44.176-18.695 46.106-42.563l26.008-321.601h6.543A25.434 25.434 0 00413.39 137.3a25.434 25.434 0 003.484-22.86zM167.555 30h83.062v28.977h-83.062zm162.804 437.02c-.68 8.402-7.796 14.98-16.203 14.98H109.293c-8.406 0-15.523-6.578-16.203-14.98L67.273 147.836h288.899zM31.793 117.836l9.27-27.79c.21-.64.808-1.07 1.484-1.07h333.082c.676 0 1.27.43 1.484 1.07l9.27 27.79zm0 0"
            />
            <path
              d="M282.516 465.957c.265.016.527.02.793.02 7.925 0 14.55-6.211 14.964-14.22L312.36 181.36c.43-8.273-5.93-15.332-14.199-15.761-8.293-.442-15.328 5.925-15.762 14.199l-14.082 270.398c-.43 8.274 5.926 15.332 14.2 15.762zm0 0M120.566 451.793c.438 7.996 7.055 14.184 14.965 14.184.274 0 .555-.008.832-.024 8.27-.45 14.61-7.52 14.16-15.793L135.77 179.762c-.45-8.274-7.52-14.614-15.793-14.16-8.27.449-14.61 7.52-14.16 15.793zm0 0M209.254 465.977c8.285 0 15-6.715 15-15V180.578c0-8.285-6.715-15-15-15s-15 6.715-15 15v270.399c0 8.285 6.715 15 15 15zm0 0"
            />
          </svg>
        </div>
      </div>
    </div>
  </span>
  <div
    className="c23"
  >
    <div
      className="c24"
    >
      <div
        className="c25"
      >
        <div>
          <input
            checked={true}
            id="isAvailable"
            onChange={[Function]}
            style={
              Object {
                "height": 0,
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "top": 0,
                "visibility": "hidden",
              }
            }
            type="checkbox"
          />
          <label
            htmlFor="isAvailable"
          >
            <div
              style={
                Object {
                  "background": "#71737e",
                  "borderRadius": 20,
                  "boxSizing": "border-box",
                  "cursor": "pointer",
                  "height": 24,
                  "padding": 2,
                  "transition": "background 200ms ease-in-out",
                  "width": 44,
                }
              }
            >
              <div
                style={
                  Object {
                    "background": "#FFF",
                    "borderRadius": "50%",
                    "height": 20,
                    "transform": "translateX(20px)",
                    "transition": "transform 200ms ease-in-out",
                    "width": 20,
                  }
                }
              />
            </div>
          </label>
        </div>
        <div
          className="c26"
        >
          <label
            className="c27"
            htmlFor="isAvailable"
          >
            Make it Available
          </label>
          <span
            className="c28"
          >
            This Template is currently available for Editors
          </span>
        </div>
      </div>
      <div
        className="c29"
        id="removePage"
        onClick={[Function]}
      >
        <svg
          style={
            Object {
              "fill": "#FFF",
              "height": "20px",
              "width": "20px",
            }
          }
          viewBox="-47 0 512 512"
        >
          <path
            d="M416.875 114.441L405.57 80.555a31.527 31.527 0 00-29.941-21.578h-95.012V28.043C280.617 12.582 268.047 0 252.59 0h-87.008c-15.453 0-28.027 12.582-28.027 28.043v30.934H42.547a31.528 31.528 0 00-29.945 21.578L1.297 114.44a25.426 25.426 0 003.484 22.856 25.427 25.427 0 0020.578 10.539h11.817L63.184 469.44C65.117 493.305 85.367 512 109.293 512h204.863c23.922 0 44.176-18.695 46.106-42.563l26.008-321.601h6.543A25.434 25.434 0 00413.39 137.3a25.434 25.434 0 003.484-22.86zM167.555 30h83.062v28.977h-83.062zm162.804 437.02c-.68 8.402-7.796 14.98-16.203 14.98H109.293c-8.406 0-15.523-6.578-16.203-14.98L67.273 147.836h288.899zM31.793 117.836l9.27-27.79c.21-.64.808-1.07 1.484-1.07h333.082c.676 0 1.27.43 1.484 1.07l9.27 27.79zm0 0"
          />
          <path
            d="M282.516 465.957c.265.016.527.02.793.02 7.925 0 14.55-6.211 14.964-14.22L312.36 181.36c.43-8.273-5.93-15.332-14.199-15.761-8.293-.442-15.328 5.925-15.762 14.199l-14.082 270.398c-.43 8.274 5.926 15.332 14.2 15.762zm0 0M120.566 451.793c.438 7.996 7.055 14.184 14.965 14.184.274 0 .555-.008.832-.024 8.27-.45 14.61-7.52 14.16-15.793L135.77 179.762c-.45-8.274-7.52-14.614-15.793-14.16-8.27.449-14.61 7.52-14.16 15.793zm0 0M209.254 465.977c8.285 0 15-6.715 15-15V180.578c0-8.285-6.715-15-15-15s-15 6.715-15 15v270.399c0 8.285 6.715 15 15 15zm0 0"
          />
        </svg>
      </div>
    </div>
    <div
      className="c24"
    >
      <div>
        <input
          checked={true}
          id="allowEdit"
          onChange={[Function]}
          style={
            Object {
              "height": 0,
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
              "visibility": "hidden",
            }
          }
          type="checkbox"
        />
        <label
          htmlFor="allowEdit"
        >
          <div
            style={
              Object {
                "background": "#71737e",
                "borderRadius": 20,
                "boxSizing": "border-box",
                "cursor": "pointer",
                "height": 24,
                "padding": 2,
                "transition": "background 200ms ease-in-out",
                "width": 44,
              }
            }
          >
            <div
              style={
                Object {
                  "background": "#FFF",
                  "borderRadius": "50%",
                  "height": 20,
                  "transform": "translateX(20px)",
                  "transition": "transform 200ms ease-in-out",
                  "width": 20,
                }
              }
            />
          </div>
        </label>
      </div>
      <div
        className="c26"
      >
        <label
          className="c27"
          htmlFor="allowEdit"
        >
          Allow editing the Page Template
        </label>
      </div>
    </div>
  </div>
</div>
`;
